include/devicetree/pwms.h

```diff
@@ -0,0 +1,383 @@
+ *
+ * @param node_id node identifier
+ * @param cell binding's cell name within the specifier at index "idx"
+ * @param idx logical index into the property
+ * @return the value of the cell inside the specifier at index "idx"
+ * @see DT_PHA_PHANDLE_IDX()
+ */
+#define DT_PWMS_CELL_BY_IDX(node_id, cell, idx) \
+	DT_PHA_BY_IDX(node_id, pwms, idx, cell)
+
+/**
+ * @brief Get pwm controller "cell" value by name
+ *
+ * Example devicetree fragment:
+ *
+ *     pwm1: pwm-controller@... {
+ *             label = "PWM_1";
+ *             #pwm-cells = <2>;
+ *     };
+ *
+ *     n: node {
+ *             pwms = <&pwm1 1 PWM_POLARITY_NORMAL>,
+ *                    <&pwm2 3 PWM_POLARITY_INVERTED>;
+ *             pwm-names = "alpha", "beta";
+ *     };
+ *
+ * Bindings fragment for the pwm1 node:
+ *
+ *     pwm-cells:
+ *       - channel
+ *       - flags
+ *
+ * Example usage:
+ *
+ *     DT_PWMS_CELL_BY_NAME(DT_NODELABEL(n), alpha, channel) // 1
+ *     DT_PWMS_CELL_BY_NAME(DT_NODELABEL(n), beta, flags) // PWM_POLARITY_INVERTED
+ *
+ * @param node_id node identifier
+ * @param name lowercase-and-underscores specifier name
+ * @param cell binding's cell name within the specifier referenced as "name"
+ * @return the value of the cell inside the named specifier
+ * @see DT_PHA_PHANDLE_IDX()
+ */
+#define DT_PWMS_CELL_BY_NAME(node_id, name, cell) \
+	DT_PHA_BY_NAME(node_id, pwms, name, cell)
+
+/**
+ * @brief Equivalent to DT_PWMS_CELL_BY_IDX(node_id, cell, 0)
+ * @param node_id node identifier
+ * @param cell binding's cell name within the specifier at index 0
+ * @return the value of the cell inside the specifier at index 0
+ * @see DT_PWMS_CELL_BY_IDX()
+ */
+#define DT_PWMS_CELL(node_id, cell) DT_PWMS_CELL_BY_IDX(node_id, cell, 0)
+
+/**
+ * @brief Get pwm controller 'channel' at an index
+ *
+ * This macro only works for PWM controllers that specify a 'channel'
+ * field in the phandle-array specifier.  Refer to the specific PWM
+ * controller binding if needed.
+ *
+ * @param node_id node identifier
+ * @param idx logical index into the property
+ * @return the channel value for the named specifier at index idx
+ * @see DT_PHA_BY_IDX()
+ * @see DT_PHA()
+ */
+#define DT_PWMS_CHANNEL_BY_IDX(node_id, idx) \
+	DT_PWMS_CELL_BY_IDX(node_id, channel, idx)
+
+/**
+ * @brief Get PWM controller 'channel' by name
+ *
+ * This macro only works for PWM controllers that specify a 'channel'
+ * field in the phandle-array specifier.  Refer to the specific PWM
+ * controller binding if needed.
+ *
+ * @param node_id node identifier
+ * @param name lowercase-and-underscores "pwm" name
+ * @return the channel value for the named specifier by name
+ * @see DT_PWMS_CELL_BY_NAME()
+ */
+#define DT_PWMS_CHANNEL_BY_NAME(node_id, name) \
+	DT_PWMS_CELL_BY_NAME(node_id, name, channel)
+
+/**
+ * @brief Equivalent to DT_PWMS_CHANNEL_BY_IDX(node_id, 0)
+ * @param node_id node identifier
+ * @return the pin value for the named specifier at index idx
```

Comment:
"the channel value at index 0" ?

Reply:
fixed

Reply:
The replacement "the channel value for the named specifier at index 0" is not correct. There is no named specifier, because this is accessed by index.